webapi/middleware.go

```diff
@@ -71,7 +71,7 @@ func (s *Server) requireAdmin(c *gin.Context) {
 // downstream handlers to make use of.
 func withDcrdClient(dcrd rpc.DcrdConnect) gin.HandlerFunc {
 	return func(c *gin.Context) {
-		client, hostname, err := dcrd.Client(c)
+		client, hostname, err := dcrd.Client()
```

Comment:
I think this was good before. is `*gin.Context` doing something weird?

Reply:
I see. Why do you need to connect to dcrd over and over, could just stay connected? Anyway, could run the context from the server I think.

Reply:
`dcrd.Client()` makes use of [`rpc.client.dial()`](https://github.com/jholdstock/vspd/blob/d8dd02433c38eed51779fd84a46ac627b4ab14c8/rpc/client.go#L85-L109), which will return an existing connection if one exists. It only creates new connections if required.